### ISSUE TITLE
Suppressed Bot cannot unmute itself

### ISSUE THREAD

**Issue by DeveloperTK** [DV8FromTheWorld/JDA#1781]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

  
- [x] I have checked for similar issues.
- [x] I have updated to the [latest JDA version][download].
- [x] I have checked the branches or the maintainers' PRs for upcoming bug fixes.

## Bug Report

The bot sometimes cannot unmute itself when it is being suppressed in an AFK channel.

### Expected Behavior

When queueing `Member#mute(false)`, the bot unmutes itself

### Code Example or Reproduction Steps

1. Let the bot join the AFK Channel
2. Try unmuting the Bot inside of the `GuildVoiceJoinEvent`

### Code for JDABuilder or DefaultShardManagerBuilder Used

```java
JDABuilder.createDefault(token).build()
```

### Exception or Error

N/A

Some things I have experiencing while trying to solve this problem: Every time the bot was already server muted (not suppressed), it could unmute itself just fine. I have also got this working by manually calling the API like this: (using OkHttp)
```java
Request request = new Request.Builder()
                .url("https://discord.com/api/v9/guilds/" + guildId + "/members/" + memberId)
                .patch(RequestBody.create(MediaType.get("application/json"), "{\"mute\": false}"))
                .header("Authorization", "Bot " + token)
                .build();
```

Could this maybe have something to do with JDA caching the muted value and refusing to update the muted state? I have got a callback while queueing the `Member#mute()` action so I suppose that is not the problem here...

**Comment by Andre601** [comment]
I think that is normal?
Last time I entered an afk channel as a owner was I muted too and couldn't unmute myself, so I would assume it's the same for bots

**Comment by arynxd** [comment]
Just tested Andre601's theory, can confirm the owner cant speak in the AFK channel.

**Comment by DeveloperTK** [comment]
Yes exactly that is why my plan is to unmute the bot inside the `GuildVoiceJoinEvent` which does not work.

**Comment by DManstrator** [comment]
I guess this is some kind of intended behavior.

Yes, you would need to remove the server mute first but the self-mute still remains and I don't think bots can remove that mute if they have it like normal user can. Perhaps this is a Discord Bug that this self-mute also applies to Bots when they join the AFK channel, not sure (I didn't test it).

Other possibility: You first join the channel and then you will be guild-muted. And basically what happens is that you try to unmute before the mute happend, leading to no effect.

From the docs of `Guild#mute`:
> Note: The Member's GuildVoiceState.isGuildMuted() value won't change until JDA receives the GuildVoiceGuildMuteEvent event related to this change.

So I would wait for that event to kick in before you try to unmute or just leave the channel / close the audio connection if the bot was moved to the AFK channel.

**Comment by DeveloperTK** [comment]
If that was true then muting and unmuting the bot with a fixed delay of a few seconds should have worked (something I tried yesterday). I have also tried unmuting the bot from the `GuildVoiceGuildMuteEvent` which did not work.

> Other possibility: You first join the channel and then you will be guild-muted. And basically what happens is that you try to unmute before the mute happend, leading to no effect.

I doubt that this is the problem here (at least on the Discord side of things) since unmuting the member manually via an API call as soon as the member joins works just fine. As I said in my first post, it could have something to do with JDA not making that call because it does not know the member was supressed inside the AFK channel and won't make the api call to unmute them (please check this I have no idea where to look in the code)

**Comment by DeveloperTK** [comment]
This check seems to be the problem. JDA does not consider the user being suppressed and only checks if the internal GuildMuted state would be unaffected by the API call.
```Java
if (voiceState.isGuildMuted() == mute)
    return new CompletedRestAction<>(getJDA(), null);
```

**Comment by DV8FromTheWorld** [comment]
Issue was fixed in #1783
